test/integration/reasoner/resolution/ResolverManagerTest.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2021 Grakn Labs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package grakn.core.reasoner.resolution;
+
+import grakn.core.common.parameters.Arguments;
+import grakn.core.concept.ConceptManager;
+import grakn.core.concept.type.EntityType;
+import grakn.core.concept.type.RelationType;
+import grakn.core.logic.LogicManager;
+import grakn.core.logic.resolvable.Concludable;
+import grakn.core.logic.resolvable.Resolvable;
+import grakn.core.logic.resolvable.Retrievable;
+import grakn.core.pattern.Conjunction;
+import grakn.core.pattern.Disjunction;
+import grakn.core.rocks.RocksGrakn;
+import grakn.core.rocks.RocksSession;
+import grakn.core.rocks.RocksTransaction;
+import grakn.core.test.integration.util.Util;
+import graql.lang.Graql;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Set;
+
+import static grakn.common.collection.Collections.list;
+import static grakn.common.collection.Collections.set;
+import static junit.framework.TestCase.assertEquals;
+
+public class ResolverManagerTest {
+
+    private static Path directory = Paths.get(System.getProperty("user.dir")).resolve("resolver-manager-test");
+    private static String database = "resolver-manager-test";
+    private static RocksGrakn grakn;
+    private static RocksSession session;
+    private static RocksTransaction rocksTransaction;
+    private static ConceptManager conceptMgr;
+    private static LogicManager logicMgr;
+
+    @Before
+    public void setUp() throws IOException {
+        Util.resetDirectory(directory);
+        grakn = RocksGrakn.open(directory);
+        grakn.databases().create(database);
+        session = grakn.session(database, Arguments.Session.Type.SCHEMA);
+        rocksTransaction = session.transaction(Arguments.Transaction.Type.WRITE);
+        conceptMgr = rocksTransaction.concepts();
+        logicMgr = rocksTransaction.logic();
+    }
+
+    @After
+    public void tearDown() {
+        rocksTransaction.close();
+        session.close();
+        grakn.close();
+    }
+
+    private Conjunction parse(String query) {
+        return Disjunction.create(Graql.parsePattern(query).asConjunction().normalise()).conjunctions().iterator().next();
```

Comment:
hurr ok, I think you'll want to apply the `TypeResolver.resolve(conjunction)` here to be consistent with how reasoning will execute: otherwise it may not find applicable rules? i dont quite remember and i just wrote the code hmm

Reply:
Ahhh yes you're right!